plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui.test/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/flexible/FlexDeployPreferencesPanelTest.java

```diff
@@ -17,30 +17,178 @@
+  public void testGetHelpContextId() throws InterruptedException {
+    FlexDeployPreferencesPanel panel = createPanel(true /* requireValues */);
 
     assertEquals(
         "com.google.cloud.tools.eclipse.appengine.deploy.ui.DeployAppEngineFlexProjectContext",
         panel.getHelpContextId());
   }
+
+  @Test
+  public void testDefaultAppYamlPathSet() throws InterruptedException {
+    FlexDeployPreferencesPanel panel = createPanel(true /* requireValues */);
+
+    Text appYamlField = findAppYamlField(panel);
+    assertEquals("src/main/appengine/app.yaml", appYamlField.getText());
+    assertFalse(hasValidationError(panel));
+  }
+
+  @Test
+  public void testAppYamlPathValidation() throws InterruptedException {
+    FlexDeployPreferencesPanel panel = createPanel(true /* requireValues */);
+
+    Text appYamlField = findAppYamlField(panel);
+    appYamlField.setText("non/existing/app.yaml");
+    assertTrue(hasValidationError(panel));
+  }
+
+  @Test
+  public void testAppYamlPathValidation_noValidationIfRequireValuesIsFalse()
+      throws InterruptedException {
+    FlexDeployPreferencesPanel panel = createPanel(false /* requireValues */);
+
+    Text appYamlField = findAppYamlField(panel);
+    appYamlField.setText("non/existing/app.yaml");
+    assertFalse(hasValidationError(panel));
+  }
+
+  @Test
+  public void testAppYamlPathValidation_absoluatePathWorks() throws InterruptedException {
+    FlexDeployPreferencesPanel panel = createPanel(false /* requireValues */);
+    Text appYamlField = findAppYamlField(panel);
+
+    IPath absoluatePath = project.getLocation().append("src/main/appengine/app.yaml");
+    assertTrue(absoluatePath.isAbsolute());
+
+    appYamlField.setText(absoluatePath.toString());
+    assertFalse(hasValidationError(panel));
+  }
+
+  private FlexDeployPreferencesPanel createPanel(boolean requireValues)
+      throws InterruptedException {
+    FlexDeployPreferencesPanel panel = new FlexDeployPreferencesPanel(shellResource.getShell(),
+        project, loginService, layoutHandler, requireValues, projectRepository);
+    findAccountSelector(panel).selectAccount("user@example.com");
+    panel.latestGcpProjectQueryJob.join();
+    return panel;
+  }
+
+  private static boolean hasValidationError(FlexDeployPreferencesPanel panel) {
+    DataBindingContext context = panel.getDataBindingContext();
+    for (Object provider : context.getValidationStatusProviders()) {
+      IObservableValue value = ((ValidationStatusProvider) provider).getValidationStatus();
+      IStatus status = (IStatus) value.getValue();
+      if (!status.isOK()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private static Text findAppYamlField(Composite panel) {
+    return (Text) findControl(panel, new Predicate<Control>() {
+      @Override
+      public boolean apply(Control control) {
+        return control instanceof Text && APP_YAML_FIELD_TOOLTIP.equals(control.getToolTipText());
```

Comment:
I'm surprised there's no ID on the control, though this is a very common omission in UI toolkits. This works well enough for a test, I suppose.